### PR TITLE
Update logging to use key value pairs

### PR DESCRIPTION
**Pull request by tej99**
# Motivation and Context

Using the Godaddy logging package we can now use key value pairs
with .with(). This PR updates all logging to use key value pairs.

# What has changed

updates all logging to use key value pairs.

# How to test?

Ran ITs and ATs.
Make sure logs lines have not been missed.

# Links

### CODE REVIEW

File: src/main/java/uk/gov/ons/ctp/response/action/service/ActionPlanService.java [hunk 1]
```diff
@@ -37,25 +37,25 @@ public ActionPlanService(final ActionPlanRepository actionPlanRepo) {
 
   @CoverageIgnore
   public List<ActionPlan> findActionPlansBySelectors(final HashMap<String, String> selectors) {
-    log.debug("Finding action plans by selectors, selectors: {}", selectors);
+    log.with(selectors).debug("Finding action plans by selectors selectors");
```
Comment (ryangrundy7) [inline src/main/java/uk/gov/ons/ctp/response/action/service/ActionPlanService.java]: Repeating `selectors`

File: src/main/java/uk/gov/ons/ctp/response/action/scheduled/distribution/ActionDistributor.java [hunk 2]
```diff
@@ -99,18 +98,16 @@ private void processActions(
...
         processAction(action, requestCount, cancelCount);
       } catch (final Exception e) {
-        log.error(
-            "Could not processing action {}."
-                + " Processing will be retried at next scheduled distribution",
-            action.getId(),
-            e);
+        log.with("action_id", action.getId())
+            .error(
+                "Could not processing action. "
```
Comment (ryangrundy7) [inline src/main/java/uk/gov/ons/ctp/response/action/scheduled/distribution/ActionDistributor.java]: Not your review comment but this doesn't sound right

File: src/main/java/uk/gov/ons/ctp/response/action/client/CaseSvcClientService.java [hunk 3]
```diff
@@ -228,8 +218,7 @@ public CreatedCaseEventDTO createNewCaseEvent(
         result = objectMapper.readValue(responseBody, CreatedCaseEventDTO.class);
       } catch (final IOException e) {
         final String msg = String.format("cause = %s - message = %s", e.getCause(), e.getMessage());
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/ctp/response/action/client/CaseSvcClientService.java]: msg now unused

File: src/main/java/uk/gov/ons/ctp/response/action/client/CollectionExerciseClientService.java [hunk 4]
```diff
@@ -62,12 +64,11 @@ public CollectionExerciseDTO getCollectionExercise(final UUID collectionExcercis
         result = objectMapper.readValue(responseBody, CollectionExerciseDTO.class);
       } catch (final IOException e) {
         final String msg = String.format("cause = %s - message = %s", e.getCause(), e.getMessage());
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/ctp/response/action/client/CollectionExerciseClientService.java]: msg unused

File: src/main/java/uk/gov/ons/ctp/response/action/client/PartySvcClientService.java [hunk 5]
```diff
@@ -98,19 +99,15 @@ private PartyDTO makePartyServiceRequest(final UriComponents uriComponents) {
 
     final ResponseEntity<String> responseEntity =
         restTemplate.exchange(uriComponents.toUri(), HttpMethod.GET, httpEntity, String.class);
-    log.debug("responseEntity is {}", responseEntity);
+    log.with(responseEntity).debug("Party service response");
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/ctp/response/action/client/PartySvcClientService.java]: `.with("response_entity"...`

File: src/main/java/uk/gov/ons/ctp/response/action/endpoint/ActionRuleEndpoint.java [hunk 6]
```diff
@@ -89,8 +89,7 @@
       final @RequestBody @Valid ActionRulePostRequestDTO actionRulePostRequestDTO,
       final BindingResult bindingResult)
       throws CTPException, InvalidRequestException {
-    log.debug(
-        "Entering createActionRule with actionRulePostRequestDTO {}", actionRulePostRequestDTO);
+    log.debug("Entering createActionRule");
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/ctp/response/action/endpoint/ActionRuleEndpoint.java]: we're logging less info than we were before

File: src/main/java/uk/gov/ons/ctp/response/action/scheduled/ReportScheduler.java [hunk 7]
```diff
@@ -35,21 +35,23 @@
   @PostConstruct
   public void init() {
     reportDistributedInstanceManager.incrementInstanceCount(DISTRIBUTED_OBJECT_KEY_INSTANCE_COUNT);
-    log.info(
-        "{} {} instance/s running",
-        reportDistributedInstanceManager.getInstanceCount(DISTRIBUTED_OBJECT_KEY_INSTANCE_COUNT),
-        DISTRIBUTED_OBJECT_KEY_INSTANCE_COUNT);
+    log.with(
+            DISTRIBUTED_OBJECT_KEY_INSTANCE_COUNT,
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/ctp/response/action/scheduled/ReportScheduler.java]: is this key snake_case?

File: src/main/java/uk/gov/ons/ctp/response/action/scheduled/ReportScheduler.java [hunk 8]
```diff
@@ -35,21 +35,23 @@
...
   @PreDestroy
   public void cleanUp() {
     reportDistributedInstanceManager.decrementInstanceCount(DISTRIBUTED_OBJECT_KEY_INSTANCE_COUNT);
     reportDistributedLockManager.unlockInstanceLocks();
-    log.info(
-        "{} {} instance/s running",
-        reportDistributedInstanceManager.getInstanceCount(DISTRIBUTED_OBJECT_KEY_INSTANCE_COUNT),
-        DISTRIBUTED_OBJECT_KEY_INSTANCE_COUNT);
+    log.with(
+            DISTRIBUTED_OBJECT_KEY_INSTANCE_COUNT,
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/ctp/response/action/scheduled/ReportScheduler.java]: the key is snake_case?

File: src/main/java/uk/gov/ons/ctp/response/action/scheduled/distribution/ActionDistributor.java [hunk 9]
```diff
@@ -99,18 +98,16 @@ private void processActions(
...
       } catch (final Exception e) {
-        log.error(
-            "Could not processing action {}."
-                + " Processing will be retried at next scheduled distribution",
-            action.getId(),
-            e);
+        log.with("action_id", action.getId())
+            .error(
+                "Could not process action. "
+                    + "Processing will be retried at next scheduled distribution",
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/ctp/response/action/scheduled/distribution/ActionDistributor.java]: No need to split this string over two lines

File: src/main/java/uk/gov/ons/ctp/response/action/service/ActionPlanService.java [hunk 10]
```diff
@@ -64,18 +64,16 @@ public ActionPlan findActionPlanByName(final String name) {
       readOnly = false,
       timeout = TRANSACTION_TIMEOUT)
   public ActionPlan createActionPlan(final ActionPlan actionPlan) {
-    log.debug(
-        "Creating action plan, Name: {}, Selectors: {}",
-        actionPlan.getName(),
-        actionPlan.getSelectors());
+    log.with(actionPlan.getSelectors())
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/ctp/response/action/service/ActionPlanService.java]: need a key name

File: src/main/java/uk/gov/ons/ctp/response/action/service/ActionPlanService.java [hunk 11]
```diff
@@ -64,18 +64,16 @@ public ActionPlan findActionPlanByName(final String name) {
...
+        .debug("Creating action plan");
 
     ActionPlan savedActionPlan = saveActionPlan(actionPlan);
 
-    log.debug(
-        "Successfully created action plan, Name: {}, ActionPlanId: {}, Selectors: {}",
-        actionPlan.getName(),
-        actionPlan.getId(),
-        actionPlan.getSelectors());
+    log.with(actionPlan.getSelectors())
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/ctp/response/action/service/ActionPlanService.java]: need to supply the key

File: src/main/java/uk/gov/ons/ctp/response/action/service/ActionReportService.java [hunk 12]
```diff
@@ -16,6 +16,6 @@
   public void createReport() {
     log.debug("Entering createReport...");
     final boolean reportResult = actionReportRepository.miStoredProcedure();
-    log.debug("Just ran the mi report and result is {}", reportResult);
+    log.with("report", reportResult).debug("Just ran the mi report and result");
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/ctp/response/action/service/ActionReportService.java]: maybe delete " and result"

File: src/main/java/uk/gov/ons/ctp/response/action/service/ActionRuleService.java [hunk 13]
```diff
@@ -74,7 +74,7 @@ public ActionRule updateActionRule(final ActionRule actionRule) {
     }
 
     if (needsUpdate) {
-      log.debug("updating action with {}", existingActionRule);
+      log.with(existingActionRule).debug("updating action");
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/ctp/response/action/service/ActionRuleService.java]: name the key

File: src/main/java/uk/gov/ons/ctp/response/action/service/ActionService.java [hunk 14]
```diff
@@ -103,14 +105,15 @@ public Action findActionById(final UUID actionId) {
       readOnly = false,
       timeout = TRANSACTION_TIMEOUT)
   public List<Action> cancelActions(final UUID caseId) throws CTPException {
-    log.debug("Entering cancelAction with {}", caseId);
+    log.with(caseId).debug("Entering cancelAction");
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/ctp/response/action/service/ActionService.java]: name the key

File: src/main/java/uk/gov/ons/ctp/response/action/service/ActionService.java [hunk 15]
```diff
@@ -204,28 +207,26 @@ public void createScheduledActions(Integer actionPlanJobId) {
...
-      log.debug("newSituation = {}", newSituation);
       if (newSituation != null) {
         needsUpdate = true;
         existingAction.setSituation(newSituation);
       }
 
       if (needsUpdate) {
         existingAction.setUpdatedDateTime(DateTimeUtil.nowUTC());
-        log.debug("updating action with {}", existingAction);
+        log.with(existingAction).debug("updating action");
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/ctp/response/action/service/ActionService.java]: name the key

File: src/main/java/uk/gov/ons/ctp/response/action/service/FeedbackService.java [hunk 16]
```diff
@@ -67,18 +67,17 @@ public void acceptFeedback(final ActionFeedback feedback) throws CTPException {
...
         }
       } else {
-        log.error(
-            "Feedback Service unable to decipher the outcome {}"
-                + " from feedback - ignoring this feedback",
-            feedback.getOutcome());
+        log.with("outcome", feedback.getOutcome())
+            .error(
+                "Feedback Service unable to decipher the outcome from feedback"
+                    + " - ignoring this feedback");
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/ctp/response/action/service/FeedbackService.java]: does it need to be split over different lines?

File: src/main/java/uk/gov/ons/ctp/response/action/service/decorator/context/BusinessActionRequestContextFactory.java [hunk 17]
```diff
@@ -101,7 +101,7 @@ private void setPartiesForBusiness(ActionRequestContext context) {
       if (childParty != null) {
         childParties.add(childParty);
       } else {
-        log.info("Unable to get party with id, {}", id);
+        log.with("part_id", id).info("Unable to get party with id");
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/ctp/response/action/service/decorator/context/BusinessActionRequestContextFactory.java]: party_id

File: src/test/java/uk/gov/ons/ctp/response/action/endpoint/ActionRuleEndpointIT.java [hunk 18]
```diff
@@ -141,8 +142,12 @@ public void updateCreatedActionRule() throws UnirestException {
   }
 
   private ActionPlanDTO createActionPlan() throws UnirestException {
+    HashMap<String, String> selectors = new HashMap<>();
+    selectors.put("selector_key", "selector_value");
```
Comment (NickGrantONS) [inline src/test/java/uk/gov/ons/ctp/response/action/endpoint/ActionRuleEndpointIT.java]: what is this change?

File: src/test/java/uk/gov/ons/ctp/response/action/endpoint/ActionRuleEndpointIT.java [hunk 19]
```diff
@@ -141,8 +142,12 @@ public void updateCreatedActionRule() throws UnirestException {
...
+
     ActionPlanPostRequestDTO actionPlanDto = new ActionPlanPostRequestDTO();
     actionPlanDto.setName("notification2");
+    actionPlanDto.setSelectors(selectors);
```
Comment (NickGrantONS) [inline src/test/java/uk/gov/ons/ctp/response/action/endpoint/ActionRuleEndpointIT.java]: seems unrelated to this PR

File: src/main/java/uk/gov/ons/ctp/response/action/scheduled/ReportScheduler.java (hunk 7 above)
Comment (tej99) [reply src/main/java/uk/gov/ons/ctp/response/action/scheduled/ReportScheduler.java]: `DISTRIBUTED_OBJECT_KEY_INSTANCE_COUNT = "reportscheduler"` not quite snake case but this is the actual name given to the instance

File: src/main/java/uk/gov/ons/ctp/response/action/scheduled/ReportScheduler.java (hunk 8 above)
Comment (tej99) [reply src/main/java/uk/gov/ons/ctp/response/action/scheduled/ReportScheduler.java]: `DISTRIBUTED_OBJECT_KEY_REPORT_LATCH = "reportlatch"` not quite snake case but this is the actual name given to the instance

File: src/main/java/uk/gov/ons/ctp/response/action/service/ActionPlanService.java (hunk 10 above)
Comment (tej99) [reply src/main/java/uk/gov/ons/ctp/response/action/service/ActionPlanService.java]: `.with()` handles the object so will show up in logs as `collection_exercise_id = uuid active_enrolment = true` is it worth putting "selectors" as a key?

File: src/test/java/uk/gov/ons/ctp/response/action/endpoint/ActionRuleEndpointIT.java (hunk 18 above)
Comment (tej99) [reply src/test/java/uk/gov/ons/ctp/response/action/endpoint/ActionRuleEndpointIT.java]: whoops added this in to check what passing objects into `.with()` looks like, will remove.

File: src/test/java/uk/gov/ons/ctp/response/action/endpoint/ActionRuleEndpointIT.java (hunk 19 above)
Comment (tej99) [reply src/test/java/uk/gov/ons/ctp/response/action/endpoint/ActionRuleEndpointIT.java]: and this ^^

File: src/main/java/uk/gov/ons/ctp/response/action/service/ActionPlanService.java (hunk 10 above)
Comment (tej99) [reply src/main/java/uk/gov/ons/ctp/response/action/service/ActionPlanService.java]: Never mind going to put the key in, the keys in the selectors object are not guaranteed to be  collection_exercise_id and active_enrolment.

File: src/main/java/uk/gov/ons/ctp/response/action/endpoint/ActionRuleEndpoint.java (hunk 6 above)
Comment (tej99) [reply src/main/java/uk/gov/ons/ctp/response/action/endpoint/ActionRuleEndpoint.java]: actionRulePostRequestDTO added

File: src/main/java/uk/gov/ons/ctp/response/action/scheduled/ReportScheduler.java (hunk 7 above)
Comment (tej99) [reply src/main/java/uk/gov/ons/ctp/response/action/scheduled/ReportScheduler.java]: changed to `.with("instance_key", DISTRIBUTED_OBJECT_KEY_INSTANCE_COUNT)` and `.with("instance_count",...`

File: src/main/java/uk/gov/ons/ctp/response/action/scheduled/ReportScheduler.java (hunk 8 above)
Comment (tej99) [reply src/main/java/uk/gov/ons/ctp/response/action/scheduled/ReportScheduler.java]: same as the other one ^^

File: src/main/java/uk/gov/ons/ctp/response/action/client/CollectionExerciseClientService.java [hunk 20]
```diff
@@ -61,13 +63,11 @@ public CollectionExerciseDTO getCollectionExercise(final UUID collectionExcercis
...
       } catch (final IOException e) {
-        final String msg = String.format("cause = %s - message = %s", e.getCause(), e.getMessage());
-        log.error(msg);
-        log.error("Stacktrace: ", e);
+        log.error("Unable to read collection exercise response", e);
       }
     }
 
-    log.info("made call to collection Exercise and retrieved {}", result);
+    log.debug("made call to collection Exercise and retrieved {}", result);
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/ctp/response/action/client/CollectionExerciseClientService.java]: should use key-value pair

Comment (tej99) [reply src/main/java/uk/gov/ons/ctp/response/action/client/CollectionExerciseClientService.java]: Thanks 😅
